pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -385,6 +387,9 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
         Message<T> message;
         try {
             message = incomingMessages.poll(timeout, unit);
+            if (message != null) {
```

Comment:
nit: it doesn't seem we need `message != null` check here. because `startMessageTracking` already checks that.